src/libstore/meson.build

```diff
@@ -0,0 +1,458 @@
+project('nix-store', 'cpp',
+  version : run_command('cat', './.version', check : true).stdout().strip(),
+  default_options : [
+    'cpp_std=c++2a',
+    # TODO(Qyriad): increase the warning level
+    'warning_level=1',
+    'debug=true',
+    'optimization=2',
+    'errorlogs=true', # Please print logs for tests that fail
+  ],
+)
+
+cxx = meson.get_compiler('cpp')
+
+deps_private = [ ]
+deps_public = [ ]
+deps_other = [ ]
+
+configdata = {
+  # TODO rename, because it will conflict with downstream projects
+  'PACKAGE_VERSION' : '"' + meson.project_version() + '"',
+  'SYSTEM': '"' + host_machine.system() + '"',
+}
+
+nix_util = dependency('nix-util', required : false)
+nix_util_from_subproject = not nix_util.found()
+if nix_util_from_subproject
+  nix_util = subproject('libutil').get_variable('nix_util')
+  # subproject sadly no good for pkg-config module
```

Comment:
... which then means you can simply do:

```meson
nix_util = dependency('nix-util')
nix_util_from_subproject = nix_util.type_name() == 'internal'
```
and since libutil is processed before libstore, it will be already overridden by the time you look it up here, and it will just use the subproject version.

Also re: knowing if it's from a subproject, I thought that since nix-util has been processed via pkgconfig.generate() before being processed by declare_dependency, meson is supposed to trace that and reuse that information via the pkgconfig module inside libstore. I could be misremembering, but I thought it was supposed to work.

Reply:
I assume though that it is in fact required. The previous `required: false` was because you wanted to manually code the fallback.